fastlane/lib/fastlane/actions/download_dsyms.rb

```diff
@@ -34,6 +34,14 @@ def self.run(params)
         end
       end
 
+      def self.download(url)
+        uri = URI.parse(url)
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl = (uri.scheme == "https")
+        res = http.get(uri.request_uri)
+        res.body
+      end
```

Comment:
If you go that way, you probably want to consider the surroundings (retry mechanisms, proxies, etc).  Also you can start adding things like progress bar etc :) 

Note that each tool might use their own client (e.g. supply depends on google api client's choices - which are changing independently of you).